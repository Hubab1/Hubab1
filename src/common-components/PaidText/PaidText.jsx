import React from 'react';
import PropTypes from 'prop-types';
import GreenCheckIcon from 'common-components/GreenCheckIcon/GreenCheckIcon';
import { P } from 'assets/styles';

export const PaidText = ({ margin = '0' }) => (
    <P bold color="#56BA82" margin={margin}>
        <GreenCheckIcon />
        Paid
    </P>
);

PaidText.propTypes = {
    margin: PropTypes.string,
};

export default PaidText;
