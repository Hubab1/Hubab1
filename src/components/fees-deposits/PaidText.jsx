import React from 'react';

import GreenCheckIcon from 'components/common/GreenCheckIcon';
import { P } from 'assets/styles';


export const PaidText = ({margin="0"}) => <P bold color="#56BA82" margin={margin}><GreenCheckIcon/>Paid</P>;

export default PaidText;