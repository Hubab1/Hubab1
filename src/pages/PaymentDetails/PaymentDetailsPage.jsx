import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'constants/constants';
import captureRoute from 'utils/captureRoute';

import { fetchPayments } from 'reducers/payments';

import { PaymentDetailsCard } from 'common-components/PaymentDetails/PaymentDetailsCard/PaymentDetailsCard';
import { PaymentTerms } from 'common-components/PaymentTerms/PaymentTerms';
import { H1, P, SpacedH3 } from 'assets/styles';
import { termsDiv, viewPaymentTerms } from './styles';

export const PaymentDetailsPage = ({ profile, configuration, payables, fetchPayments, applicant }) => {
    const [currentPage, setCurrentPage] = useState('summary');
    useEffect(() => {
        fetchPayments();
    }, [fetchPayments]);

    if (!profile || !configuration || !profile.lease_start_date) {
        return null;
    }

    const unitNumber = profile.unit?.unit_number;
    const communityName = profile.community?.display_name;
    const leaseStartDate = profile.lease_start_date;
    const hasReceipt = applicant && applicant.receipt;
    const holdingDepositAmount =
        parseFloat(profile.selected_rental_options?.['holding-deposit']?.[0]?.quoted_deposit_amount) || 0;

    if (currentPage === 'terms') {
        return (
            <>
                <PaymentTerms
                    handleClickBack={() => setCurrentPage('summary')}
                    holdingDepositAmount={holdingDepositAmount}
                    unitNumber={unitNumber}
                    communityName={communityName}
                    leaseStartDate={leaseStartDate}
                    canProceedToPayment={false}
                />
            </>
        );
    } else {
        return (
            <>
                <H1>Payment Details</H1>
                <SpacedH3>Hereâ€™s the breakdown for unit {unitNumber}</SpacedH3>
                <PaymentDetailsCard fetchPayments={fetchPayments} payables={payables} profile={profile} />
                {hasReceipt && (
                    <div className={termsDiv}>
                        <P fontSize={14}>
                            <span role="button" onClick={() => setCurrentPage('terms')} className={viewPaymentTerms}>
                                Payment and Holding Deposit Terms
                            </span>
                        </P>
                    </div>
                )}
            </>
        );
    }
};

PaymentDetailsPage.propTypes = {
    fetchPayments: PropTypes.func.isRequired,
    profile: PropTypes.object.isRequired,
    applicant: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    payables: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    configuration: state.configuration,
    payables: state.payments || [],
    applicant: state.applicant,
});

export default connect(mapStateToProps, { fetchPayments })(captureRoute(PaymentDetailsPage, ROUTES.PAYMENT_DETAILS));
