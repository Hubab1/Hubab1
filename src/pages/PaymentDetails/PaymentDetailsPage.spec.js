import React from 'react';
import { shallow, mount } from 'enzyme';

import { PaymentDetailsPage } from './PaymentDetailsPage';
import PaidText from 'common-components/PaidText/PaidText';

describe('PaymentDetails', () => {
    let defaultProps;

    beforeEach(() => {
        defaultProps = {
            profile: {
                lease_start_date: '2020-12-15',
                unit: {
                    unit_number: 1,
                },
                lease_term: 12,
                fees_breakdown: {
                    application_fees: {
                        total: '620',
                        items: [],
                    },
                    move_in_fees: {
                        total: '0',
                        items: [],
                    },
                    monthly_fees: {
                        total: '1234',
                        items: [],
                    },
                },
            },
            configuration: {},
            fetchPayments: jest.fn(),
            applicationFees: {
                total: 620,
                allPaid: true,
                items: [
                    {
                        name: 'Application Fee',
                        amount: 500,
                        quantity: 4,
                        price: 125,
                        type: 'fee',
                    },
                    {
                        name: 'Holding Deposit',
                        amount: 120,
                        quantity: 1,
                        price: 120,
                        type: 'fee',
                    },
                ],
            },
        };
    });

    it('renders the correct number of rows', () => {
        const wrapper = shallow(<PaymentDetailsPage {...defaultProps} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders the "Paid" text if all due at application payments have been paid', () => {
        const wrapper = mount(<PaymentDetailsPage {...defaultProps} />);

        expect(defaultProps.fetchPayments).toBeCalledTimes(1);
        expect(wrapper.find(PaidText).length).toEqual(1);
    });

    it('does not render the "Paid" text if applicationFees.allPaid is false', () => {
        const applicationFees = { ...defaultProps.applicationFees, allPaid: false };

        const wrapper = mount(<PaymentDetailsPage {...{ ...defaultProps, applicationFees }} />);

        expect(defaultProps.fetchPayments).toBeCalledTimes(1);
        expect(wrapper.find(PaidText).length).toEqual(0);
    });

    it('displays terms link when passed payment', () => {
        const applicant = {
            receipt: { id: 123 },
        };
        const wrapper = shallow(<PaymentDetailsPage {...defaultProps} applicant={applicant} />);
        expect(wrapper.text()).toContain('Payment and Holding Deposit Terms');
    });

    it('Does not display terms link when payment is incomplete/pending', () => {
        const applicant = {
            receipt: null,
        };
        const wrapper = shallow(<PaymentDetailsPage {...defaultProps} applicant={applicant} />);
        expect(wrapper.text()).not.toContain('Payment and Holding Deposit Terms');
    });
});
