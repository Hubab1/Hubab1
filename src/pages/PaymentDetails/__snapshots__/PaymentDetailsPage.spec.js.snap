// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDetails renders the correct number of rows 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Payment Details
  </ForwardRef(render)>
  <ForwardRef(render)>
    Hereâ€™s the breakdown for unit 
    1
  </ForwardRef(render)>
  <PaymentDetailsCard
    applicationFees={
      Object {
        "allPaid": true,
        "items": Array [
          Object {
            "amount": 500,
            "name": "Application Fee",
            "price": 125,
            "quantity": 4,
            "type": "fee",
          },
          Object {
            "amount": 120,
            "name": "Holding Deposit",
            "price": 120,
            "quantity": 1,
            "type": "fee",
          },
        ],
        "total": 620,
      }
    }
    fetchPayments={[MockFunction]}
    profile={
      Object {
        "fees_breakdown": Object {
          "application_fees": Object {
            "items": Array [],
            "total": "620",
          },
          "monthly_fees": Object {
            "items": Array [],
            "total": "1234",
          },
          "move_in_fees": Object {
            "items": Array [],
            "total": "0",
          },
        },
        "lease_start_date": "2020-12-15",
        "lease_term": 12,
        "unit": Object {
          "unit_number": 1,
        },
      }
    }
  />
</React.Fragment>
`;
